src/subcommand/init/shell/powershell.rs

```diff
@@ -0,0 +1,84 @@
+use super::{HookConfig, ShellConfig};
+
+use anyhow::Result;
+
+use std::borrow::Cow;
+
+pub const CONFIG: ShellConfig = ShellConfig {
+    z,
+    alias,
+    hook: HookConfig {
+        prompt: HOOK_PROMPT,
+        pwd: hook_pwd,
+    },
+};
+
+fn z(z_cmd: &str) -> String {
+    format!(
+        r#"
+function {} {{
+    function z_cd($dir) {{
+        try {{
+            Set-Location $dir -ea Stop
+            if ($env:_ZO_ECHO -eq "1") {{
+                Write-Host "$PWD"
+            }}
+        }} catch {{
+        }}
+    }}
+
+    if ($args.Length -eq 0) {{
+        z_cd ~
+    }}
+    elseif ($args.Length -eq 1 -and $args[0] -eq '-') {{
+        z_cd -
+    }}
+    else {{
+        $result = zoxide query @args
+        if ($LASTEXITCODE -eq 0 -and ((Test-Path $result) -eq $true)) {{
+            z_cd $result
+        }} else {{
+            $result
+        }}
+    }}
+}}
+"#,
+        z_cmd
+    )
+}
+
+fn alias(z_cmd: &str) -> String {
+    format!(
+        r#"
+function zi {{ {} -i @args }}
```

Comment:
The previous commit changes the aliases when `cmd` is changed. For example, this alias would be written as:

```
function {}i {{ {} -i @args }}
```